portfolio/src/main/java/com/google/sps/servlets/CommentRepresentationSerializer.java

```diff
@@ -1,20 +1,33 @@
 package com.google.sps.servlets;
 
+import java.lang.reflect.Field;
 import java.util.ArrayList;
 
 public class CommentRepresentationSerializer {
   public String username;
   public String text;
   public String key;
+  public String timestamp;
   public String replyTo;
   // the idea here is to have a list of ids of comments
   // that are direct replies to this comment
   public ArrayList<String> replies;
 
   public CommentRepresentationSerializer(Comment comment) {
-    this.username = comment.username;
-    this.text = comment.text;
     this.replies = new ArrayList<>();
-    this.key = comment.key;
+    for (Field field: comment.getClass().getDeclaredFields()) {
+      // here I just copy each property of comment to the serializer
+      // it's okay that some fields may not be initialized -
+      // serializer makes its best to initialize itself
+      try {
+        String fieldName = field.getName();
+        String fieldValue = (String)field.get(comment);
+        if (fieldValue != null) {
+          CommentRepresentationSerializer.class.getField(fieldName).set(this, fieldValue);
+        }
+      } catch (IllegalAccessException | NoSuchFieldException ex) {
+        System.out.println(ex.getMessage());
+      }
+    }
```

Comment:
Why do we need reflection for construction after adding the new field `timestamp`, but not before?

Reply:
We need it in general, I just wrote it here.

Reply:
I mean the idea that I need some generic way to serialize all `Comment`'s fields came to me when I was adding timestamp.

Reply:
Improved it using inheritance